image-registry/automate-pruning.md

```diff
@@ -0,0 +1,325 @@
+    reason: Scheduled
+    message: "Image pruner job has been scheduled."
+  - type: Failed
+    staus: "False"
+    lastTransitionTime: 2019-10-09T03:13:45
+    reason: Succeeded
+    message: "Most recent image pruning job succeeded."
+```
+
+#### Spec
+
+1. `schedule:` Cron formatted schedule, defaults to daily at midnight for new
+clusters. Required field
+2. `suspend:` if `true`, CronJob running pruning is suspended. Optional,
+   default false.
+3. `keepTagRevisions:` number of revisions per tag to keep. Optional, default 3
+   if not set.
+4. `keepYoungerThan:` retain images younger than this duration. Optional,
+   default 60m if not set.
+5. `resources:` standard Pod resource requests and limits. Optional.
+6. `affinity:` standard Pod affinity. Optional.
+7. `nodeSelector:` standard Pod node selector. Optional
+8. `tolerations:` standard Pod tolerations. Optional
+9. `startingDeadlineSeconds:` start deadline for `CronJob`. Optional.
+10. `history:` sets the job history to be retained on the cluster. Object is
+    optional.
+    1. `successfulJobsHistoryLimit:` maximum number of successful jobs to
+       retain. Must be >= 1 to ensure metrics are reported.
+    2. `failedJobsHistoryLimit:` maximum number of failed jobs to retain. Must
+       be >= 1 to ensure metrics are reported.
+
+#### Status
+
+1. `observedGeneration:`: generation observed by the operator
+2. `conditions:` standard condition objects with the following types:
+   1. `Available` - indicates if the pruning job has been created.
+   Reasons can be `Ready` or `Error`.
+   2. `Scheduled` - indicates if the next pruning job has been scheduled.
+   Reasons can be `Scheduled`, `Suspended`, or `Error`.
+   3. `Failed` - indicates if the most recent pruning job failed.
+
+### Fixed Configuration
+
+The following settings for the pruning `CronJob` cannot be configured:
+
+1. `concurrencyPolicy: Forbid` - this prevents concurrent cron jobs from
+running.
+2. `suspend` - based on the provided `schedule`. If empty `suspend = true`.
+3. Pruning CLI:
+   1. `--prune-registry` - always `true` (default)
+   2. `--all` - always `true` (default)
+   3. `--prune-over-size-limit` - not set
+   4. `--confirm` - always `true` so pruning runs
+
+### Telemetry and Alerting
+
+1. Send to Telemeter a metric indicating if pruning has been scheduled.
+2. Fire a warning alert if pruning is suspended or could not be scheduled.
+3. Fire a warning alert if the active pruning job exceeds 24 hours in duration.
+4. Fire a warning alert if the most recent pruning job failed.
+
+### Test Plan
+
+- `e2e-*-operator` suite can verify that the `CronJob` is configured properly
+  on the cluster.
+- A separate suite can also verify that the pruning works end to end:
+  - Create a build with imagestream tag. Run the build.
+  - Delete build & IST.
+  - Run pruning.
+  - Verify image is pruned.
+
+### Graduation Criteria
+
+#### Tech Preview
+
+Requirements for reaching tech preview:
```

Comment:
you're not targeting TP right? should this section be removed/updated to be NA?

Reply:
correct - particularly since TP adds additional engineering costs